components/style/properties/declaration_block.rs

```diff
@@ -631,6 +632,13 @@ impl ToCss for PropertyDeclarationBlock {
                 // Step 3.3.1 is done by checking already_serialized while
                 // iterating below.
 
+                // If the set of previously serialized shorthands contains any of the shorthands
+                // that are associated with the current longhand, we can skip it since its value
+                // has already been included in the serialization of the related shorthand.
+                if shorthands.iter().any(|s| shorthands_used.iter().any(|used| used == s)) {
```

Comment:
Isn't this quadratic? :(

Reply:
Can we instead use something like `PropertyDeclarationIdSet`, but for shorthands instead?